Translation tools

Lanny opened the pull request:
## Description

Adds a script (`tt.js`) that generates translation templates (CVS files for translators to fill out) and ingests completed or partially completed templates back into the translation files.

Documentation and CI workflow pending

robin-macpherson commented:
This is gonna be sweeeet!

Lanny commented:
Ok, finally ready for review and merge! @robin-macpherson 

If you're good with it, we should start directing users to here: http://translations-website.s3-website.us-east-2.amazonaws.com/ for translations and move away from the google sheet (maybe revoke the ability to edit it?). I started moving the Spanish translations over. It's faster than editing the JSON files, but still a bit tedious. We should probably take responsibility for transferring what's in the google sheet but it's going to be a bit of a slog, doesn't need to be done all at once.

robin-macpherson commented on packages/web/tt.js:
```diff
@@ -0,0 +1,543 @@
+    </p>
+    <p>
+      To translate a language that isn't listed, drop us a line at
+      <a href="mailto:hello@journaly.com">hello@journally.com</a>.
```
```suggestion
      <a href="mailto:hello@journaly.com">hello@journaly.com</a>.
```

robin-macpherson commented on packages/web/tt.js:
```diff
@@ -0,0 +1,543 @@
+    <p>
+      To help translate one of the existing languages, find it in the table
+      below, download the linked template, fill it out and email it to
+      <a href="mailto:hello@journaly.com">hello@journally.com</a>.
```
```suggestion
      <a href="mailto:hello@journaly.com">hello@journaly.com</a>.
```

Lanny replied:
Oh god, I'm still doing it!

robin-macpherson commented on packages/web/tt.js:
```diff
@@ -0,0 +1,543 @@
+      <h1>Help Translate Journaly!</h1>
+    </header>
+    <p>
+      Journaly is currently accepting translations in the ${LOCALES.length}
```
```suggestion
      Journaly is currently accepting translations in ${LOCALES.length}
```

Lanny replied:
Oh god, I'm still doing it!

robin-macpherson commented on packages/web/tt.js:
```diff
@@ -0,0 +1,543 @@
+      contains the original English text, and enter the translation in the
+      "Translated String" column. Some strings contain variables which are
+      represented as "{{variable_name}}", which will be substituted with a
+      value on the site. For example the read time string is
```
Something to think about is that this particular doc is likely to be read by a lot of non-developers and also non-native English speakers, so I think the term "string" could be confusing for some people. It is in an important column name, though, so I'm wondering it maybe we should just add a little note to clarify for people just in case?

robin-macpherson commented on packages/web/tt.js:
```diff
@@ -0,0 +1,543 @@
+      <a href="mailto:hello@journaly.com">hello@journally.com</a>.
+      and we'll look at adding it. Since not every language has the same usage
+      among our userbase, and because each translation requires maintainance
+      effort, we may hold on adding some languages until we find there is a
```
```suggestion
      effort, we may wait to add some languages until we find there is a
```

(this is another tiny detail but taking out some idiomatic usage might make it more understandable for non-native English speakers reading it)

robin-macpherson commented on packages/web/tt.js:
```diff
@@ -0,0 +1,543 @@
+    </p>
+    <p>
+      Also feel free to review translations that are marked as being up to date.
+      We assume good faith and do not independently verify every translation,
```
```suggestion
      We generally trust the submissions we get and do not independently verify every translation,
```

(this is another small suggestion just to make it easier to understand for non-natives)

robin-macpherson commented on packages/web/tt.js:
```diff
@@ -0,0 +1,543 @@
+      to the discresion of translators to determine what's appropriate. As a
+      general guide, our English copy is usually written in a semi-casual
+      tone, within the UI we <em>usually</em> avoid slang, but allow
+      contractions and casual constructs. We try to make coppy clearly
```
```suggestion
      contractions and casual constructs. We try to make copy clearly
```

robin-macpherson commented on packages/web/tt.js:
```diff
@@ -0,0 +1,543 @@
+      In your translations, first consider what is natural for web content in
+      your language. If your language has strict distinctions between registers,
+      use whatever register is typical for websites. If there aren't hard and
+      fast rules in your language, try to replicate the "semi-casual" tone from
```
Maybe "if there aren't strict rules in your language"? (again just thinking about non-natives)

robin-macpherson reviewed (changes requested):
This is awesome stuff man! Super excited to make this process much much easier for folks translating and for us (my God it's been painful putting these in up to this point and especially keeping track of things going out of date or gaps that were found and communicating that clearly to translators and vice versa.

I made a few suggestions around simplifying certain idiomatic language that might be hard for non-native English speakers to understand and a few small typos. My main concern now is just, how do we expect people to open and edit this file? For me, after downloading a template file, by default the only program my mac new to open it with was VimR.

I think it's very possible/likely that a lot of people won't know how to work with CSV's and will likely end up with something like this with TextEdit if they try to open it:
![image](https://user-images.githubusercontent.com/34203886/116924706-1877f880-ac0d-11eb-81f7-3f8ee2da96cd.png)

I think it would be greatly helpful for us to provide some guidance on how best to open and work with the files. Even just something like explaining that they can open it in a spreadsheet program like Microsoft Excel or if they don't have that they can upload it to Google Sheets for free, make their changes, and then download the file as a .csv to send back to us.

What do you think?

Lanny commented:
Good points! I appreciate the thought toward non-native English speakers reading it, since that's exactly who we want to be doing translations. I added a terminology section at the start and added a paragraph about CSV editors, as well as applying your inline suggestions. Take a look whenever you get a chance to feel free to suggest more changes there as you feel is appropriate.

robin-macpherson reviewed (approved):
Awesome man! Thanks for applying all the suggestions 🙏🏼✨ Time to see this in action!
